src/Utils/InstrumentSchema.php

```diff
@@ -258,20 +253,34 @@ public static function check_field_permissions( $source, array $args, AppContext
 		) {
 
 			/**
-			 * If the schema for the field is configured to "isPrivate" or has "auth" configured,
-			 * make sure the user is authenticated before resolving the field
+			 * Retrieve permissions error message.
 			 */
-			if ( empty( get_current_user_id() ) ) {
-				throw new UserError( $auth_error );
-			}
+			$auth_error = is_array( $field->config['auth'] ) && ! empty( $field->config['auth']['errorMessage'] )
+				? $field->config['auth']['errorMessage']
```

Comment:
I think the auth error message should be a callback, with some context passed to it. 

Because the error message might need some dynamic input to do things like `sprintf( 'You are not allowed to see %s', $field_name )` or whatever. 

Like, this is a super basic example, but the error message will likely be more useful if it can provide some dynamic context about why the error occurred, and passing context to a callback can allow that.

So, if I were configuring this, I'd want: 

```php
register_graphql_field( 'RootQuery', 'testField', [
  'type' => 'String',
  'auth' => [
    'errorMessage' => function( $some_context_passed_in ) {
       return sprintf( 'You are not allowed to see the "%s" field', $field_name_from_the_context_passed_in );
    }
  ]
] );
```

Reply:
Actually, similar to how we're doing `call_user_func()` here: https://github.com/wp-graphql/wp-graphql/pull/1938/files#diff-88202b0372659f22fdbdc5fcf28bc69c356ddbadb177abf2e93ee434108051b2R268

I think we can allow String or Callable. 

So you could do: 

```php
register_graphql_field( 'RootQuery', 'testField', [
  'type' => 'String',
  'auth' => [
    'errorMessage' => 'Some static error message',
  ]
] );
```

Or: 

```php
register_graphql_field( 'RootQuery', 'testField', [
  'type' => 'String',
  'auth' => [
    'errorMessage' => function( $field, $field_key, $source, $args, $context, $info, $field_resolver ) {
       return sprintf( 'You are not allowed to see the "%s" field', field_key );
    }
  ]
] );
```

Reply:
Meh, this is something we can add support for later